docs/test_instructions_per_dependency_update.md

```diff
@@ -0,0 +1,859 @@
+    5. [PlayServicesCoreScanner](#playservicescodescanner)
+5. Network
+    1. [Okio](#okio)
+6. Tool
+    1. [Zendesk](#zendesk)
+    2. [JSoup](#jsoup)
+7. Other Core
+    1. [AutoService](#autoservice)
+    2. [KotlinPoet](#kotlinpoet)
+8. Other UI
+    1. [Lottie](#lottie)
+    2. [UCrop](#ucrop)
+9. [Smoke Test](#smoke-test)
+10. [Special](#special)
+
+ℹ️ Every test instruction should be prefixed with one of the following:
+- [JP/WP] This test applies to both, the `Jetpack` and `WordPress` apps.
+- [JP] This test applies to the `Jetpack` app only.
+- [WP] This test applies to the `WordPress` app only.
+
+-----
+
+### Navigation [[navigationVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/settings.gradle#L1)] <a name="navigation"></a>
+
+<details>
+    <summary>1. [JP/WP] Image Editing Flow [libs:image-editor]</summary>
+
+- Add a new `blog` post.
+- Add a new `image` block.
+- Choose an image and wait for it to be uploaded within the `image` block.
+- Click on the `media options` of this image (top right) and then click `edit`.
+- Verify that the `Edit Image` screen is shown and functioning as expected.
+- Crop the image and click the `done` menu option (top right).
+- Make sure the image is updated accordingly.
+
+</details>
+
+-----
+
+### WorkManager [[androidxWorkManagerVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/build.gradle#L54)] <a name="workmanager"></a>
+
+<details>
+  <summary>1. [JP/WP] UploadWorker.kt</summary>
+
+- Go to `Post` screen.
+- Create a new post and publish it.
+- Turn device offline.
+- Go to this post and update it.
+- Notice the warning message: `We'll publish the post when your device is back online.`
+- Turn device online.
+- Notice this post being automatically uploaded.
+- Open this post on a web browser and verify the post is indeed updated.
+
+</details>
+
+<details>
+  <summary>2. [JP] ReminderWorker.kt</summary>
+
+- Go to `Site Settings` screen.
+- Find the `Blogging` section, click on `Reminders`, toggle-on every day and click on `Update`.
+- Notice the `All set!` bottom sheet appearing, click `Done`.
+- Close the app, preferably swipe the app off.
+- Go to the device's `Settings` app, find the `Date & Time` section, turn `Automatic date & time`
+  off.
+- Set the device's date to a day after today.
+- Open the app.
+- Verify the `Blogging Reminders` notification appearing. For example, the notification title could
+  be `Daily Prompt`, while the notification description something like `Is there anything you feel
+  too old to do anymore?`.
+
+</details>
+
+<details>
+  <summary>3. [JP] LocalNotificationWorker.kt</summary>
+
+- TODO
+- TODO
+- TODO
+
+</details>
+
+<details>
+  <summary>4. [JP] WeeklyRoundupWorker.kt</summary>
+
+- TODO
+- TODO
+- TODO
+
+</details>
+
+-----
+
+### Preference [[androidxPreferenceVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/build.gradle#L50)] <a name="preference"></a>
```

Comment:
Yea, you're right about that @irfano and I was wondering how to go best about it, just because of the pending `android.preference` to `androidx.preference` migration... 🤔 

What about instead of removing the how-to-test section and thus all those testing scenarios, to instead add an higher level info within that `Preference` section explaining that fact, including links to the migration issue, `android.preference` to `androidx.preference`, but still keep those testing scenarios, wdyt? 🤔

Reply:
I would prefer to remove those screens because we don't know when to migrate to `androidx.preference`. Screens and features may change until we migrate, and it could be pointless to keep these test cases until we migrate. However, I understand your point. You think that these cases are already written and do not lose them. 
So, I leave the decision to you; both options are fine with me.

Reply:
👋 @irfano , wow about this 2f1403d9a7efc9e3cde2a4ff6e53815e789c7d19 change that I pushed just now? 🤔

Reply:
Better now. 😊 Thanks for the change.

Reply:
Awesome, thank YOU for helping me doing this change, YOU rock! 🤟